### PR TITLE
Datasource description

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
* Added description field to datasource metadata.
* Removed restriction on all fields with prefix auth. Now only username, password, access_key, secret_key are the only fields removed from GET datasource API.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/main/java/org/opensearch/sql/datasource/model/DataSourceMetadata.java [hunk 1]
```diff
@@ -11,26 +11,29 @@
...
 @Setter
 @AllArgsConstructor
-@NoArgsConstructor
 @EqualsAndHashCode
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class DataSourceMetadata {
 
   @JsonProperty private String name;
 
+  @JsonProperty private String description;
```
Comment (derek-ho) [inline core/src/main/java/org/opensearch/sql/datasource/model/DataSourceMetadata.java:35]: Does this annotation provide default getter/setter? Do not see code for that

Comment (vamsi-amazon) [reply core/src/main/java/org/opensearch/sql/datasource/model/DataSourceMetadata.java:35]: There is getter setter annotation above class name. JsonProperty is usually not required unless we have some customization in serialization like changing the name of the field to something other than field name/ignoring in serialization.

**Review by dai-chen — APPROVED** [review]
Just to confirm the BWC, old data source metadata can still be handled by this new code?

**Comment by vamsi-amazon** [comment]
> Just to confirm the BWC, old data source metadata can still be handled by this new code?

Yes it can handle, BWC is because of different issue.

**Pull request merged**